Snowflake.Data.Tests/UnitTests/SFAzureClientTest.cs

```diff
@@ -0,0 +1,245 @@
+﻿/*
+ * Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests.UnitTests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Core;
+    using Snowflake.Data.Core.FileTransfer.StorageClient;
+    using Snowflake.Data.Core.FileTransfer;
+    using System.Collections.Generic;
+    using System.Net;
+    using Snowflake.Data.Tests.Mock;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.IO;
+
+    [TestFixture]
+    class SFAzureClientTest : SFBaseTest
+    {
+        // Mock data for file metadata
+        const string EndPoint = "blob.core.windows.net";
+
+        const string LocationStage = "mock-customer-stage";
+        const string LocationId = "mock-id";
+        const string LocationTables = "tables";
+        const string LocationKey = "mock-key";
+        const string LocationPath = LocationTables + "/" + LocationKey + "/";
+        const string Location = LocationStage + "/" + LocationId + "/" + LocationPath;
+
+        const string Region = "canadacentral";
+
+        Dictionary<string, string> _stageCredentials = new Dictionary<string, string>()
+        {
+            {"AZURE_SAS_TOKEN", "MOCK_AZURE_SAS_TOKEN"}
+        };
+
+        const string StorageAccount = "mockStorageAccount";
+
+        // Settings for mock client
+        const int Parallel = 0;
+
+        // File name for download tests
+        const string DownloadFileName = "mockFileName.txt";
+
+        // Token for async tests
+        CancellationToken _cancellationToken;
+
+        // Mock upload file size
+        const int UploadFileSize = 9999;
+
+        // The mock client and metadata
+        SFSnowflakeAzureClient _client;
+        SFFileMetadata _fileMetadata;
+
+        [SetUp]
+        public void BeforeTest()
+        {
+            _fileMetadata = new SFFileMetadata()
+            {
+                stageInfo = new PutGetStageInfo()
+                {
+                    endPoint = EndPoint,
+                    isClientSideEncrypted = true,
+                    location = Location,
+                    locationType = SFRemoteStorageUtil.AZURE_FS,
+                    path = LocationPath,
+                    presignedUrl = null,
+                    region = Region,
+                    stageCredentials = _stageCredentials,
+                    storageAccount = StorageAccount
+                }
+            };
+
+            // Setup mock client
+            MockAzureClient mockClient = new MockAzureClient();
```

Comment:
To me we are testing a wrapper over a mocked blob service. Does it make sense to rename MockAzureClient to sth like MockAzureBlobClient or MockAzureBlobService.

Reply:
That makes sense, I'll rename it to `MockAzureBlobClient `